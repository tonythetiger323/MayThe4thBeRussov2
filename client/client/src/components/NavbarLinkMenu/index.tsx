import * as React from 'react'
import Button from '@material-ui/core/Button'
import Menu from '@material-ui/core/Menu'
import MenuItem from '@material-ui/core/MenuItem'
import NavbarLink from '../NavbarLink'
import PopupState, {
  bindTrigger,
  bindMenu,
} from 'material-ui-popup-state/index';

const NavbarLinkMenu: React.SFC = () => {
  return (
    <PopupState variant='popover' popupId='NavbarLink-popup-menu'>
      {(popupState: any) => (
        <React.Fragment>
          <Button {...bindTrigger(popupState)}>Their Story</Button>
          <Menu {...bindMenu(popupState)}>
            <MenuItem onClick={popupState.close}>
              <NavbarLink to='/theirstory'> Their Story</NavbarLink>
            </MenuItem>
            <MenuItem onClick={popupState.close}>
              <NavbarLink to='/herstory'>Her Story</NavbarLink>
            </MenuItem>
            <MenuItem onClick={popupState.close}>
              <NavbarLink to='/hisstory'>His Story</NavbarLink>
            </MenuItem>
          </Menu>
        </React.Fragment>
      )}
    </PopupState>
  );
};

export default NavbarLinkMenu;
