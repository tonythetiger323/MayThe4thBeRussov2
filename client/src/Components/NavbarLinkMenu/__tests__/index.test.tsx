import * as React from 'react'
import { shallow } from 'enzyme'
import NavbarLinkMenu from '../../NavbarLinkMenu'
import MenuItem from '@material-ui/core/MenuItem'
import { createShallow } from '@material-ui/core/test-utils'

let wrapper: any
let shallowMui: any

describe('NavbarLinkMenu Component', () => {
  beforeEach(() => (wrapper = shallow(<NavbarLinkMenu />)))

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})

describe('<MenuItem/>', () => {
  beforeEach(() => {
    shallowMui = createShallow()
  })

  it('should work', () => {
    wrapper = shallowMui(<MenuItem />)
  })
})

