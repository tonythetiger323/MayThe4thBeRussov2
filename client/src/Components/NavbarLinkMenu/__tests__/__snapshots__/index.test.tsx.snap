// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLinkMenu Component should render correctly 1`] = `
<PopupState
  popupId="NavbarLink-popup-menu"
  variant="popover"
>
  <Component />
</PopupState>
`;
